scheduler/src/main/java/org/apache/falcon/execution/ProcessExecutionInstance.java

```diff
@@ -120,14 +125,40 @@ private void registerForNotifications(boolean isResume) throws FalconException {
                 continue;
             }
             Feed feed = ConfigurationStore.get().get(EntityType.FEED, input.getFeed());
-            List<Path> paths = new ArrayList<>();
+            String startTimeExp = input.getStart();
+            String endTimeExp = input.getEnd();
+            DateTime processInstanceTime = getInstanceTime();
+            expressionHelper.setReferenceDate(new Date(processInstanceTime.getMillis()));
+
+            Date startTime = expressionHelper.evaluate(startTimeExp, Date.class);
+            Date endTime = expressionHelper.evaluate(endTimeExp, Date.class);
+
             for (org.apache.falcon.entity.v0.feed.Cluster cluster : feed.getClusters().getClusters()) {
+                org.apache.falcon.entity.v0.cluster.Cluster clusterEntity =
+                        EntityUtil.getEntity(EntityType.CLUSTER, cluster.getName());
+                if (!EntityUtil.responsibleFor(clusterEntity.getColo())) {
+                    continue;
+                }
+                List<Path> paths = new ArrayList<>();
                 List<Location> locations = FeedHelper.getLocations(cluster, feed);
                 for (Location loc : locations) {
                     if (loc.getType() != LocationType.DATA) {
                         continue;
                     }
-                    paths.add(new Path(loc.getPath()));
+                    List<Date> instanceTimes = EntityUtil.getEntityInstanceTimes(feed, cluster.getName(),
+                            startTime, endTime);
+                    for (Date instanceTime : instanceTimes) {
+                        String path = EntityUtil.evaluateDependentPath(loc.getPath(), instanceTime);
```

Comment:
how will latest be supported in this case? or else latest is not supported with this pull req?